### PR TITLE
move ImagePath to computed property for display

### PR DESCRIPTION
**Pull request by tehviking**
This resolves the issue where clicking back button or going to "index" threw the error 

`cannot read property 'unchain' of undefined`

### CODE REVIEW

**Comment by hone** [comment]
@tehviking thanks! can you explain why I need to move this to a computed property for display or what a computed property is?

**Comment by tehviking** [comment]
So a computed property is just a function that, when called, behaves like a property on an object.
http://emberjs.com/guides/object-model/computed-properties/

In this case we're just aliasing one property to another to change the name. That's to work around the bug in Ember that causes the "cannot read property 'unchain' of undefined" when you declare a variable in your Handlebars templates with uppercase letters. Handlebars used to reserve upper-cased words for the concept of Globals, which aren't a thing anymore, so this is a bug in the framework that's currently being resolved.

So all we did was use a computed property to alias `Front.ImagePath` to `imagePath` on the controller. This is pretty common in Ember, to clean up nested accessors. So like if you do `Ember.computed.alias("foo.bar.baz", "baz")` it'll define a property named `baz` that lets you set or get the `foo.bar.baz` property without typing it out.

TL:DR; Computed properties can do a lot more than this, but in this case we were just aliasing to get the property downcased in the Handlebars template to work around a bug in Handlebars.

**Comment by hone** [comment]
@tehviking omg. <3 <3 thanks.

**Comment by hone** [comment]
@tehviking why is the computed propery on the controller?

**Comment by tehviking** [comment]
Controllers are pretty much wrongly named in Ember. They're essentially proxies or decorators. The job of the controller is to proxy and decorate an underlying model (as well as handle actions). I sometimes describe them as a "Quesarito":



The burrito is the model, and the controller is the quesadilla that wraps it. 

If you ask a controller for a property, it'll ask itself first, but if it doesn't have the answer, it'll ask the underlying model. If you ask a model for its "model" property, it'll give you the unwrapped, original model. But otherwise, any get and set stuff you do on a controller will proxy to model it wraps.

Controllers are the place for changing names of properties for display (such as this case), holding on to additional state (think `"isSelected"`), or anything else that involves displaying and handling changes on a given model.

In this case, changing the name is only really for display, so the controller is the right place for it.

**Comment by tehviking** [comment]
Alternatively, we could have defined a model and placed the behavior on the model like this:

``` javascript
// app/models/card.js 
import Ember from 'ember';

export default Ember.Object.extend({
  imagePath: Ember.computed.alias("Front.ImagePath");
})
```

Then, instead of `Ember.Object.create(cardItem)` in the application route, we'd say 

``` javascript
import Card from '../models/card'
... 
foo.map(function(cardIem) {
  return Card.create(cardItem);
...
```

This would place the responsibility on the Card model, but in our case it was both simpler and more accurate to place that alias responsibility on the wrapping controller. 

You will, eventually, want and need a Card model as defined above to define and enforce the idea of a "Card" on the client side, but we can blindly just throw the data up on screen for now.
